zipline/pipeline/factors/factor.py

```diff
@@ -833,6 +834,84 @@ def linear_regression(self, target, regression_length, mask=NotSpecified):
             mask=mask,
         )
 
+    def winsorize(self,
```

Comment:
Clamp is slightly different than winsorize. In clamping, the upper and lower bounds are provided, in winsorizing they are derived from the data set. For example the lower limit in inclusive winsorizing in generated by arr[int(limit*len(arr))].